### PR TITLE
Enabling deep-tests on forks

### PR DESCRIPTION
**Pull request by davidcok**
<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: .github/workflows/msbuild.yml [hunk 1]
```diff
@@ -55,11 +55,11 @@ jobs:
...
     needs: check-deep-tests
-    if: always() && ((needs.check-deep-tests.result == 'success' || contains(github.event.pull_request.labels.*.name, 'run-deep-tests')) || ( github.event_name == 'push' && ( github.ref_name == 'master' || vars.TEST_ON_FORK == 'true' )))
+    if: always() && ((needs.check-deep-tests.result == 'success' || contains(github.event.pull_request.labels.*.name, 'run-deep-tests')) || ( github.event_name == 'push' && vars.TEST_ON_FORK == 'true' ))
     uses: ./.github/workflows/integration-tests-reusable.yml
     with:
       # By default run only on one platform, but run on all platforms if the PR has the "run-deep-tests"
       # label, and skip checking the nightly build above.
       # This is the best way to fix an issue in master that was only caught by the nightly build.
-      all_platforms: ${{ contains(github.event.pull_request.labels.*.name, 'run-deep-tests') }}
+      all_platforms: ${{ contains(github.event.pull_request.labels.*.name, 'run-deep-tests') || contains(github.event.push.labels.*.name, 'run-deep-tests')}}
```
Comment (keyboardDrummer) [inline .github/workflows/msbuild.yml:64]: What is the difference between these two?

File: .github/workflows/msbuild.yml [hunk 2]
```diff
@@ -55,11 +55,11 @@ jobs:
 
   integration-tests:
     needs: check-deep-tests
-    if: always() && ((needs.check-deep-tests.result == 'success' || contains(github.event.pull_request.labels.*.name, 'run-deep-tests')) || ( github.event_name == 'push' && ( github.ref_name == 'master' || vars.TEST_ON_FORK == 'true' )))
```
Comment (keyboardDrummer) [inline .github/workflows/msbuild.yml:58]: Why is this change needed?

**Review by keyboardDrummer — APPROVED** [review]

File: .github/workflows/msbuild.yml (hunk 1 above)
Comment (davidcok) [reply .github/workflows/msbuild.yml]: The change allows one to run deep tests on a local fork, with a push on a branch that does not have a PR created. Allows running full tests without using common resources.

File: .github/workflows/msbuild.yml (hunk 2 above)
Comment (davidcok) [reply .github/workflows/msbuild.yml:58]: To avoid running tests on the push to the master branch that happens after a merge of a PR that has just been tested as a PR event.

**Comment by davidcok** [comment]
Made obsolete by other edits
